tests/check/check_h26xsigned_auth.c

```diff
@@ -1625,6 +1625,228 @@ START_TEST(vendor_axis_communications_operation)
+      // one generated on the camera side)
+      signed_video_generate_private_key(
+          algo, "./", (char **)&sign_info_wrong_key.private_key, &sign_info_wrong_key.private_key_size);
+      openssl_read_pubkey_from_private_key(&sign_info_wrong_key);
+      // Set public key
+      sv_rc = signed_video_set_public_key(sv_vms, sign_info_wrong_key.public_key, sign_info_wrong_key.public_key_size);
+      ck_assert_int_eq(sv_rc, SV_OK);
+    }
+
+    if (pk_tests[j].set_public_key) {
+      // Set public key
+      sv_rc = signed_video_set_public_key(
+          sv_vms, sv_camera->signature_info->public_key, sv_camera->signature_info->public_key_size);
+      ck_assert_int_eq(sv_rc, SV_OK);
+    }
+
+    signature_info_t *sign_info;
+    if (pk_tests[j].signature_info) {
+      if (pk_tests[j].wrong_pk) {
+        sign_info = &sign_info_wrong_key;
+      } else {
+        sign_info = sv_camera->signature_info;
+      }
+    } else {
+      sign_info = NULL;
+    }
+    validation_helper_function(sv_vms, sei, pk_tests[j].wrong_pk, sign_info, codec);
+
+    signed_video_free(sv_camera);
+    free(private_key);
+  }
+}
+END_TEST
+
+/* Test description */
+START_TEST(no_public_key_in_sei_and_bad_public_key_on_validation_side)
+{
+  // This test runs in a loop with loop index _i, corresponding to struct sv_setting _i in
+  // |settings|; See signed_video_helpers.h.
+
+  SignedVideoReturnCode sv_rc;
+  SignedVideoCodec codec = settings[_i].codec;
+  sign_algo_t algo = settings[_i].algo;
+  nalu_list_item_t *i_nalu = nalu_list_item_create_and_set_id("I", 0, codec);
+  nalu_list_item_t *sei = NULL;
+  char *private_key = NULL;
+  signed_video_t *sv_camera = NULL;
+
+  // On camera side
+  sv_camera = generate_and_set_private_key_on_camera_side(settings[_i], false, i_nalu, &sei);
+
+  // On validation side
+  signed_video_t *sv_vms = signed_video_create(codec);
+
+  // Generate a new private key in order to extract a bad private key (a key not compatible with the
+  // one generated on the camera side)
+  signature_info_t sign_info = {0};
+  signed_video_generate_private_key(
+      algo, "./", (char **)&sign_info.private_key, &sign_info.private_key_size);
+  openssl_read_pubkey_from_private_key(&sign_info);
+  // Set public key
+  sv_rc = signed_video_set_public_key(sv_vms, sign_info.public_key, sign_info.public_key_size);
+  ck_assert_int_eq(sv_rc, SV_OK);
+
+  // Validate this first GOP.
+  signed_video_authenticity_t *auth_report = NULL;
+
+  sv_rc = signed_video_add_nalu_and_authenticate(sv_vms, sei->data, sei->data_size, &auth_report);
+  ck_assert_int_eq(sv_rc, SV_OK);
+  ck_assert(!auth_report);
+
+  sv_rc =
+      signed_video_add_nalu_and_authenticate(sv_vms, i_nalu->data, i_nalu->data_size, &auth_report);
+  ck_assert_int_eq(sv_rc, SV_OK);
+
+  // TODO: This test is correct but currently one I-frame is not enough. The state "ok with missing
+  // info" will be used until the bug is fixed.
```

Comment:
This comment is not really correct. Comment says "OK_WITH_MISSING_INFO", but you use "SIGNATURE_PRESENT"